optuna/samplers/_partial_fixed.py

```diff
@@ -0,0 +1,122 @@
+                y = trial.suggest_int("y", -1, 1)
+                return x ** 2 + y
+
+            study = optuna.create_study()
+            study.optimize(objective, n_trials=10)
+
+            best_params = study.best_params
+            fixed_params = {"y": best_params["y"]}
+            base_sampler = optuna.samplers.RandomSampler()
+            partial_sampler = PartialFixedSampler(fixed_params, base_sampler)
+
+            study.sampler = partial_sampler
+            study.optimize(objective, n_trials=10)
+
+    .. seealso::
+
+        Please check :class:`optuna.samplers.BaseSampler` page for
+        more information on how the sampler works.
+
+    """
+
+    def __init__(self, fixed_params: Dict[str, Any], base_sampler: BaseSampler) -> None:
+        self._fixed_params = fixed_params
+        self._base_sampler = base_sampler
+
+    def reseed_rng(self) -> None:
+
+        self._rng = numpy.random.RandomState()
+
+    def infer_relative_search_space(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+    ) -> Dict[str, BaseDistribution]:
+
+        search_space = self._base_sampler.infer_relative_search_space(study, trial)
+
+        # Remove fixed params from relative search space to return fixed values.
+        for param_name in self._fixed_params.keys():
+            if param_name in search_space:
+                del search_space[param_name]
+
+        return search_space
+
+    def sample_relative(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        search_space: Dict[str, BaseDistribution],
+    ) -> Dict[str, Any]:
+
+        # Fixed params are never sampled here.
+        return self._base_sampler.sample_relative(study, trial, search_space)
+
+    def sample_independent(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        param_name: str,
+        param_distribution: BaseDistribution,
+    ) -> Any:
+
+        # Fixed params will be sampled here.
+
+        # If param_name isn't in self._fixed_params.keys(), param_value == None.
+        param_value = self._fixed_params.get(param_name)
+
+        if param_value:
+            # Check if a parameter value is contained in the range of this distribution.
+            param_value_in_internal_repr = param_distribution.to_internal_repr(param_value)
+            contained = param_distribution._contains(param_value_in_internal_repr)
+
+            if not contained:
+                warnings.warn(
+                    "Fixed parameter '{}' with value {} is out of range "
+                    "for distribution {}.".format(param_name, param_value, param_distribution)
+                )
+            return param_value
+        else:
+            return self._base_sampler.sample_independent(
+                study, trial, param_name, param_distribution
+            )
+
+    @property
+    def fixed_params(self) -> Dict[str, Any]:
+        """Return fixed parameters.
+
+            .. versionadded:: 2.2.0
```

Comment:
IMO, we can just remove this `.. versionadded::` since the experimental decorator will be added above.
```suggestion
```